tfx/experimental/pipeline_testing/executor_verifier.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Recording pipeline from MLMD metadata."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import absl
+import os
+import types
+from typing import Text
+
+from tensorflow_metadata.proto.v0 import anomalies_pb2
+import tensorflow_model_analysis as tfma
+from tfx.utils import io_utils
+from tfx.experimental.pipeline_testing import verifier_utils
+
+class ExecutorVerifier(object):
+  """ExecutorVerifier for verifying executor outputs"""
+  def __init__(self, record_dir,
+               pipeline_info,
+               metadata_connection_config,
+               threshold=0.5):
+    """
+    threshold: between 0 and 1
+    components_id: components to verify
+    pipeline_info: for current pipeline
+    metadata_connection_config: for current pipeline
+    """
+    self._record_dir = record_dir
+    # self._metadata_connection_config = metadata_connection_config
+    # self._pipeline_info = pipeline_info
+    self._threshold = threshold
+
+    # default verifier
+    self._verifier_map = {'ExampleValidator': self.validator_verifier,
+                          'Trainer': self.trainer_verifier,
+                          'Evaluator': self.evaluator_verifier}
+    self.component_output_map = verifier_utils.get_component_output_map(
+        metadata_connection_config, pipeline_info)
+
+  def trainer_verifier(self, component_id, output_dict):
```

Comment:
I think that we can have these pre-defined verifiers as separate functions (not included in the class). For example,
```
def trainer_verifier(record_dir, output_dict):
     ...
```
And users might be able to call `verify` with the dictionary of component_id -> verifier function without set_verifier() steps. Because users should be able to write their own verifiers, these functions could serve as an example.